pkg/gatewayserver/io/udp/udp.go

```diff
@@ -69,6 +68,10 @@ func Serve(ctx context.Context, server io.Server, conn *net.UDPConn, config Conf
 	if config.RateLimiting.Enable {
 		firewall = NewRateLimitingFirewall(firewall, config.RateLimiting.Messages, config.RateLimiting.Threshold)
 	}
+	limitLogs, err := limitLogsConfig.New(ctx, limitLogsSize)
+	if err != nil {
+		panic(err)
```

Comment:
Return `err`. No need to `panic`.

Reply:
Opted to `panic` because this is coming from internal configuration, so an error message is not really helpful.

Reply:
I understand that the intent is to stop the stack, by force if necessary, if the configuration is invalid. But I'd prefer that `limitLogsSize` is validated as part of the configuration itself, instead of here. You can return the error and never start the stack at that point.

I don't think there is any reason to have `panic(err)` in production code.